### PR TITLE
Sudo - add suport for many passwords

### PR DESCRIPTION
**Pull request by marcin-usielski**

### CODE REVIEW

File: moler/cmd/unix/sudo.py [hunk 1]
```diff
@@ -191,10 +200,11 @@ def _process_wrong_password(self, line):
         :raises: ParsingDone if regex matches the line.
         """
         if re.search(self._get_wrong_password_regex(), line):
-            if self._sent_password and not self._command_output_started:
-                self.set_exception(CommandFailure(self, "Command error password found in line '{}'.".format(line)))
-                self._finish_on_final_prompt = True
-                self._sent_password = False
+            if not self._command_output_started:
+                if (len(self.password) <= 0) and (self.repeat_password is False):
```
Comment (tkrol88) [inline moler/cmd/unix/sudo.py:204]: why just `len(self.password) == 0`?

File: moler/cmd/unix/sudo.py [hunk 2]
```diff
@@ -233,7 +243,19 @@ def _parse_password(self, line):
         """
         if re.search(self._get_password_regex(), line):
             if not self._sent_password:
-                self.connection.sendline(self.password, encrypt=self.encrypt_password)
+                try:
+                    pwd = self.password.pop(0)
```
Comment (tkrol88) [inline moler/cmd/unix/sudo.py:247]: hmm, maybe different name would be more readable. At first glance I thought that you send `pwd` command :D

File: moler/cmd/unix/sudo.py (hunk 1 above)
Comment (marcin-usielski) [reply moler/cmd/unix/sudo.py]: Done

File: moler/cmd/unix/sudo.py (hunk 2 above)
Comment (marcin-usielski) [reply moler/cmd/unix/sudo.py]: Done. `current_password`.

**Pull request merged**